talpid-wireguard/src/lib.rs

```diff
@@ -601,7 +601,7 @@ impl WireguardMonitor {
     /// Reconfigures the tunnel to use the provided config while potentially modifying the config
     /// and restarting the obfuscation provider. Returns the new config used by the new tunnel.
     async fn reconfigure_tunnel(
-        tunnel: &Arc<Mutex<Option<Box<dyn Tunnel>>>>,
+        tunnel: &Arc<AsyncMutex<Option<Box<dyn Tunnel>>>>,
```

Comment:
Unrelated to your PR, but isn't this arguement type a bit weird? Why is the tunnel wrapped in both an `Arc` and `Box`? And why is the option inside the mutex, wouldn't it make more sense with a `Option<Arc<AsyncMutex<dyn Tunnel>>>`? Actually, we don't seem to mutate the tunnel in any way, why not just take a `&dyn Tunnel`? :thinking_face:

Reply:
Good questions

Reply:
This should probably be refactored at some point IMO, but let's drop it for this PR.